src/ln/channelmanager.rs

```diff
@@ -466,6 +476,30 @@ macro_rules! try_chan_entry {
 				}
 				return Err(MsgHandleErrInternal::from_finish_shutdown(msg, channel_id, chan.force_shutdown(), $self.get_channel_update(&chan).ok()))
 			},
+			Err(ChannelError::Wait { msg, update }) => {
+				log_error!($self, "Channel {} need to be shutdown but closing transactions not broadcast due to {}", log_bytes!($entry.key()[..]), msg);
+				let (channel_id, mut chan) = $entry.remove_entry();
+				if let Some(short_id) = chan.get_short_channel_id() {
+					$channel_state.short_to_id.remove(&short_id);
+				}
+				if let Some(update) = update {
+					if let Err(e) = $self.monitor.add_update_monitor(update.get_funding_txo().unwrap(), update) {
+						match e {
+							// Our local state is rotten as we're fallen-behind and we can't send our updates
+							// to watchtowers to retrieve our funds from remote commitment transaction.
+							// We close channnel as there is no hope to get back something..
+							ChannelMonitorUpdateErr::PermanentFailure => {
+								return Err(MsgHandleErrInternal::from_finish_shutdown(msg, channel_id, chan.force_shutdown(), $self.get_channel_update(&chan).ok()))
+							},
+							// We may be able to send lastest remote per commitment point to one of our watchtowers
+							// and get back some funds after temporary failure source has ceased.
+							ChannelMonitorUpdateErr::TemporaryFailure => {}
+						}
+					}
+				}
+				chan.force_shutdown();
```

Comment:
Hmmmmmm....what to do with this result is...pretty unclear. We can't just blindly drop the HTLCs we may need to fail back, but probably shouldn't immediately fail them back. Any ideas?

Reply:
Assuming we have lost data to reconstruct offered HTLC script, it's lost money anyway given we can't timeout them so best it to avoid break backward channel (if state is correct for this one, which is also unlikely, I don't see doing backup per-channel granularity). See https://github.com/lightningnetwork/lightning-rfc/blob/master/05-onchain.md#rationale-4. 
Yes best seem to fail HTLCs backward ?

Reply:
Add 16c285f for what we were talking yesterday